src/main/java/com/dataguise/saas/JWTLoginFilter.java

```diff
@@ -0,0 +1,51 @@
+package com.dataguise.saas;
+
+import com.dataguise.saas.dto.LoginDTO;
+import com.dataguise.saas.service.JWTService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Collections;
+
+public class JWTLoginFilter extends AbstractAuthenticationProcessingFilter {
+
+    public JWTLoginFilter(String url, AuthenticationManager authManager) {
+        super(new AntPathRequestMatcher(url));
+        setAuthenticationManager(authManager);
+    }
+
+    @Override
+    public Authentication attemptAuthentication(
+            HttpServletRequest req, HttpServletResponse res)
+            throws AuthenticationException, IOException, ServletException {
+        LoginDTO creds = new ObjectMapper()
+                .readValue(req.getInputStream(), LoginDTO.class);
+        return getAuthenticationManager().authenticate(
+                new UsernamePasswordAuthenticationToken(
+                        creds.getUsername(),
+                        creds.getPassword(),
+                        Collections.emptyList()
+                )
+        );
+    }
+
+    @Override
```

Comment:
**Reporter**: Checkstyle    
    **Rule**: com.puppycrawl.tools.checkstyle.checks.design.DesignForExtensionCheck
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/JWTLoginFilter.java L42
    
    Class &#39;JWTLoginFilter&#39; looks like designed for extension (can be subclassed), but the method &#39;successfulAuthentication&#39; does not have javadoc that explains how to do that safely. If class is not designed for extension consider making the class &#39;JWTLoginFilter&#39; final or making the method &#39;successfulAuthentication&#39; static/final/abstract/empty, or adding allowed annotation for the method.
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1801180249>*